dll/win32/kernel32/client/console/console.c

```diff
@@ -3067,6 +3066,48 @@ UnregisterConsoleIME(VOID)
     return FALSE;
 }
 
+/**
+ * @brief
+ * Internal helper function used to synchronize the current
+ * thread's language ID with the one from the console.
+ **/
+VOID
+SetTEBLangID(VOID)
+{
+    CONSOLE_API_MESSAGE ApiMessage;
+    PCONSOLE_GETLANGID LangIdRequest = &ApiMessage.Data.LangIdRequest;
+
+    /* Retrieve the "best-suited" language ID corresponding
+     * to the active console output code page. */
+    LangIdRequest->ConsoleHandle = NtCurrentPeb()->ProcessParameters->ConsoleHandle;
+
+    CsrClientCallServer((PCSR_API_MESSAGE)&ApiMessage,
+                        NULL,
+                        CSR_CREATE_API_NUMBER(CONSRV_SERVERDLL_INDEX, ConsolepGetLangId),
+                        sizeof(*LangIdRequest));
+    if (!NT_SUCCESS(ApiMessage.Status))
+    {
+        /*
+         * No available console language ID: keep the current thread's one.
+         * Since this internal function modifies an optional setting only,
+         * don't set any last error, as it could otherwise mess with the
+         * main last error set by the caller.
+         */
+        return;
+    }
+
+    /*
+     * We succeeded, set the current thread's language ID by
+     * modifying its locale -- Windows <= 2003 does not have
+     * the concept of a separate thread UI language.
+     * Ignore the returned value.
+     */
+    if (!SetThreadLocale(MAKELCID(LangIdRequest->LangId, SORT_DEFAULT)))
+    {
+        DPRINT1("SetTEBLangID: Could not set thread locale to console lang ID %lu\n",
+                LangIdRequest->LangId);
+    }
+}
```

Comment:
1. How to detect whether the target font is absent or not.
2. If absent, what should we have to do?

Reply:
For the font, this will be in a separate PR.